var/spack/repos/builtin/packages/netcdf-cxx4/package.py

```diff
@@ -43,7 +44,17 @@ def flag_handler(self, name, flags):
             flags.append(self.compiler.cc_pic_flag)
         elif name == 'cppflags':
             flags.append('-I' + self.spec['netcdf-c'].prefix.include)
-
+        elif name == 'ldflags':
+            # We need to specify LDFLAGS to get correct dependency_libs
+            # in libnetcdf_c++.la, so packages that use libtool for linking
+            # can correctly link to all the dependencies even when the
+            # building takes place outside of Spack environment, i.e.
+            # without Spack's compiler wrappers.
+            config_flags = [self.spec['netcdf-c'].libs.search_flags]
+            # On macOS, we also need to add this to ldflags
+            if self.spec.satisfies("platform=darwin"):
+                flags = [self.spec['netcdf-c'].libs.search_flags, 
+                         self.spec['netcdf-c'].libs.link_flags] + flags
```

Comment:
I am curious to learn why you think `link_flags` should be added. As far as I know, the configure script appends `-lnetcdf` automatically.